packages/swingset-liveslots/test/vpid-liveslots.test.js

```diff
@@ -898,3 +898,100 @@ test('inter-vat circular promise references', async t => {
   // });
   // t.deepEqual(log, []);
 });
+
+test('accept previously allocated promise', async t => {
+  function build(vatPowers, vatParameters) {
+    const { target } = vatParameters;
+    return Far('root', {
+      call() {
+        const promise = E(target).foo();
+        // wrap to avoid adoption
+        return { promise };
+      },
+      async watch(p) {
```

Comment:
let's add some comments and rename things to remind us what's happening here:
```js
  async waitFor(promise2) {
    // if we're in the same incarnation as the `E(target).foo()`, this `promise2` will be the same JS Promise as the `promise` above
```

* `promise2` to emphasize the expected equivalence (barring an upgrade)
* avoid using `watch` because we aren't talking about `VatData.watchPromise` in this test